prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirBundleUtils.kt

```diff
@@ -44,13 +44,27 @@ object FhirBundleUtils : Logging {
     fun convertFhirType(value: Base, sourceType: String, targetType: String, logger: KotlinLogger = this.logger): Base {
         return if (sourceType == targetType || targetType == "*") {
             value
+        } else if (targetType == StringCompatibleType.DateTime.typeAsString) {
+            when (sourceType) {
+                StringCompatibleType.DateTime.typeAsString -> value
+                StringCompatibleType.Instant.typeAsString -> DateTimeType(value.primitiveValue())
+                StringCompatibleType.Date.typeAsString -> DateTimeType(value.primitiveValue())
```

Comment:
Dates don't have timezones, but a date can be converted to a dateTime, because the dateTime can have a precision level of YEAR or MONTH or DAY. At those precisions, the timezone will be ignored, but either way, I think the conversion should be allowed at this level.